Fix Issue #39: Add ChIP-seq Large Margin Visualization to Gallery

ANAMASGARD opened the pull request:
Fixes #39 
Fixes #35

This PR adds the ChIP-seq large margin visualization AND fixes the title mismatch issue.

## 1. ChIP-seq Large Margin Viz (Issue #39)

### Source Code
- Repo: https://github.com/ANAMASGARD/chip-seq-large-margin-code
- Ported from old animint to animint2 syntax
- Added auto-download for required data files

### Deployed Viz
- Repo: https://github.com/ANAMASGARD/chip-seq-large-margin
- Live at: https://ANAMASGARD.github.io/chip-seq-large-margin/
- Deployed with `animint2pages()` including title and source parameters

### Gallery Update
- Added `ANAMASGARD/chip-seq-large-margin` to `repos.txt`
- Generated screenshot: `repos/ANAMASGARD/chip-seq-large-margin.png`

## 2. Gallery Title Fix (Issue #35)

Fixed title mismatch for climate visualization:
- **Before**: "Climate change sensor stations"
- **After**: "CRUTEM4 Temperature Sensor Stations"

The title now matches what's in the source code. Fixed by clearing cached metadata and re-running `update_gallery()`.

## Preview
https://ANAMASGARD.github.io/gallery/

@tdhock This is ready for review!

ANAMASGARD committed:
update 2025-12-08 15:39:28.834598

ANAMASGARD committed:
update 2025-12-19 15:47:32.761824

ANAMASGARD committed:
update 2025-12-19 15:50:16.493868

tdhock commented:
there are two issues.
old viz below has widerect which is easier to select than the segment you used.
<img width="442" height="409" alt="image" src="https://github.com/user-attachments/assets/23efa029-faa3-4fc4-8343-06fd25e6e2ae" />
your viz below has blue text too small to read.
<img width="842" height="510" alt="image" src="https://github.com/user-attachments/assets/a234a737-2254-4d31-ac45-9e252f3c9316" />
please fix. (use widerect, and increase text size)

also why are there so many diff lines in meta.csv?
There should be only 2 because you only modified two gallery entries, right?

tdhock reviewed (changes requested):
please fix

ANAMASGARD committed:
Fix gallery metadata for climate and chip-seq entries

- Fixed climate viz title to match source code (Issue #35)
- Added chip-seq large margin entry (Issue #39)
- Regenerated index.html from the updated meta.csv

ANAMASGARD commented:
Sir @tdhock,
I’ve addressed your feedback:-

- Made the chip‑seq selection areas wide/easy to click (like widerect).
- Increased the blue explanatory text and sample label sizes so they are readable.
<img width="3072" height="1920" alt="Screenshot From 2026-02-11 14-16-15" src="https://github.com/user-attachments/assets/0c014c37-9545-46f4-8c28-e958d0ad6e07" />

 - Cleaned up meta.csv so it only changes the climate title (Issue #35) and adds the chip‑seq entry (Issue #39), and regenerated index.html from that.
 
 PLease give your feedback

tdhock commented:
rect looks good but blue text still looks too small. it should be about the same size as surrounding text.

ANAMASGARD committed:
Update screenshot with larger text size both for blue and black text

ANAMASGARD commented:
Sir @tdhock  I have increased the size of both the blue text and the black made the size=8  for clear visibility.

 - Please review it (Re-Deployed) :-  https://anamasgard.github.io/chip-seq-large-margin/

<img width="3072" height="1920" alt="Screenshot From 2026-02-15 21-09-43" src="https://github.com/user-attachments/assets/29b3fcb4-1442-400e-8f9a-72ed23caa7ba" />

tdhock commented:
the text is still too small, can you please increase so that it is same as surrounding text size?

ANAMASGARD commented:
Sir @tdhock  
Increase the size made it 12 please tell , it is readable without zoom .

<img width="3072" height="1920" alt="image" src="https://github.com/user-attachments/assets/9062d97f-80a2-4bb8-8f1e-b482d117823b" />

tdhock commented:
on my screen constant and margin are unreadable (go off of right side of screen). please revise so that they are readable in plotting area.

also the original has smooth transitions: when you change peaks, segment in regression plot updates with a smooth transition. can you please add that?

your viz below only has one peaks selector
<img width="1221" height="911" alt="image" src="https://github.com/user-attachments/assets/4131827e-775c-46eb-848a-47d4f423ab6c" />

original below has a peaks selector for each sample
<img width="959" height="685" alt="image" src="https://github.com/user-attachments/assets/f1c04de0-833d-49a2-b0e4-1e8731d60898" />

2 more problems with top plot:
<img width="921" height="381" alt="image" src="https://github.com/user-attachments/assets/6035db0d-09b4-4f13-83a7-f101c0e80bcb" />

* yours only displays blue peaks for the selected sample (original displays blue peaks for all samples)
* sample names in panel titles overlap

can you please fix?

ANAMASGARD committed:
Update chip-seq screenshot

ANAMASGARD committed:
Updated chip-seq screenshot with all fixes

ANAMASGARD commented:
Sir  @tdhock, I pushed another update to the ChIP-seq viz :-

 - Moved “large margin / small margin / constant” text fully inside the regression panel (no cutoff)
 - Added smooth transitions when changing peaks
 - Peaks selection is now per-sample (4 selectors)
 - Top plot now shows blue peaks for all samples (not only the selected one)
 - Fixed the overlapping sample strip titles
Updated version: https://ANAMASGARD.github.io/chip-seq-large-margin/
I also updated the gallery screenshot in this PR. 
- But McGill0002 sample.ids and blue text they still look small should I increase the size or not ??
 
 Sir review and give your feedback .

tdhock commented:
please compare with original. yours looks like this, too many panels
<img width="839" height="716" alt="image" src="https://github.com/user-attachments/assets/dd624cad-635b-4c7b-9ea9-b8898824b206" />

ANAMASGARD commented:
Sir @tdhock, I want to explain what happened clearly and ask your preferred next step.

I made a mistake while trying to update the ChIP-seq screenshot/workflow.
I pushed directly to upstream gh-pages, and that caused the PR state to change/merge flow unexpectedly.
Then I reverted those merged changes in commit 5bb862a (Revert PR #46), so the current branch no longer contains the PR #46 content.

- The original work is not deleted from history (commits like 4b1c7ac and related PR #46 commits are still visible).
- I understand this caused confusion and extra churn, and I am sorry for that mistake I didn't merged it intentionally i just made all the changes in the 
- https://github.com/ANAMASGARD/chip-seq-large-margin-code

Sorry please tell me what should  I do now  ????

suhaani-agarwal commented:
@ANAMASGARD please do not push directly to gh-pages going forward just keep that in mind.
It’s good that you’ve reverted the changes. Please open two separate PRs to address the individual issues (#39 and #35), and please ensure that all previously requested changes are fully incorporated.
